src/opower/utilities/exelon.py

```diff
@@ -125,32 +201,14 @@ async def async_login(
         # If pepco or delmarva, determine if we should use secondary subdomain
         if cls.login_domain() in ["secure.pepco.com", "secure.delmarva.com"]:
             # Get the account type & state
-            async with session.get(
-                "https://" + cls.login_domain() + "/.euapi/mobile/custom/auth/accounts",
-                headers={
-                    "User-Agent": USER_AGENT,
-                    "authorization": f"Bearer {result['access_token']}",
-                },
-                raise_for_status=True,
-            ) as resp:
-                # returned mimetype is nonstandard, so this avoids a ContentTypeError
-                response = await resp.json(content_type=None)
-
-                # Only include active accounts (after moving, old accounts have status: "Inactive")
-                # NOTE: this logic currently assumes 1 active address per account, if multiple accounts found
-                #      we default to taking the first in the list. Future enhancement is to support
-                #      multiple accounts (which could result in different subdomain for each)
-                active_accounts = [
-                    account
-                    for account in response["data"]
-                    if account["status"] == "Active"
-                ]
-                isResidential = active_accounts[0]["isResidential"]
-                state = active_accounts[0]["PremiseInfo"][0]["mainAddress"][
-                    "townDetail"
-                ]["stateOrProvince"]
-                _LOGGER.debug("found exelon account isResidential: %s", isResidential)
-                _LOGGER.debug("found exelon account state: %s", state)
+            account = await cls.async_account(session, result["access_token"])

```

Comment:
Updated, I did the same check in both places since the lookup wasn't super close to the declaration just kept it the same, but I can remove that if you prefer.